ui/raidboss/data/04-sb/ultimate/unending_coil_ultimate.ts

```diff
@@ -1552,6 +1764,97 @@ const triggerSet: TriggerSet<Data> = {
         },
       },
     },
+    // For Grand Octet:
+    // After bosses and dragons start spawning, there's no clear log line we can trigger off of to find bosses' position
+    // until it's effectively too late.  The best way to do this seems to be to fire the trigger
+    // with a delay when Bahamut uses Grand Octet before all 3 bosses jump.
+    {
+      id: 'UCU Grand Octet Run & Rotate',
+      type: 'Ability',
+      // Grab mob position data after dragons/bosses are positioned
+      netRegex: { id: '26E7', source: 'Bahamut Prime', capture: false },
+      delaySeconds: 4.8,
+      promise: async (data) => {
+        if (!data.trioSourceIds.nael || !data.trioSourceIds.bahamut)
+          return;
+        data.combatantData = [];
+        data.combatantData = (await callOverlayHandler({
+          call: 'getCombatants',
+          ids: [data.trioSourceIds.nael, data.trioSourceIds.bahamut],
+        })).combatants;
+      },
+      alertText: (data, _matches, output) => {
+        let naelDirIdx;
+        let bahaDirIdx;
+
+        for (const mob of data.combatantData) {
+          const mobDirIdx = Math.round(4 - 4 * Math.atan2(mob.PosX, mob.PosY) / Math.PI) % 8;
+          if (mob.ID === data.trioSourceIds.nael)
+            naelDirIdx = mobDirIdx;
+          else if (mob.ID === data.trioSourceIds.bahamut)
+            bahaDirIdx = mobDirIdx;
+        }
+
+        if (!naelDirIdx || !bahaDirIdx)
+          return;
+
+        // If Bahamut spaws on a cardinal, the party goes opposite and rotates counter-clockwise; if intercardinal, clockwise.
+        // If Nael is directly opposite Bahamut, the party instead starts one directional position over (same as the rotation direction)
+        // http://clees.me/guides/ucob/
+        let rotationIdxModifier; // this is used to modify the party starting spot in directions[] if Nael is opposite Bahamut
+        let rotationPath;
+
+        const bahaDirStr = dirs[bahaDirIdx];
+        if (!bahaDirStr)
+          return;
+        if (dirCardinals.includes(bahaDirStr)) {
+          rotationIdxModifier = -1;
+          rotationPath = 'counterclockwise';
+        } else {
+          rotationIdxModifier = 1;
+          rotationPath = 'clockwise';
+        }
+
+        // start by going directly opposite Bahamut
+        let partyStartIdx = bahaDirIdx >= 4 ? bahaDirIdx - 4 : bahaDirIdx + 4;
+        // If Nael is there, instead go +1/-1 direction (depending on the rotation direction)
+        if (naelDirIdx === partyStartIdx) {
+          partyStartIdx += rotationIdxModifier;
+          // if this pushes partyStartIdx beyond the array boundary, wrap around
+          if (partyStartIdx === -1) {
+            partyStartIdx = 7;
+          } else if (partyStartIdx === 8) {
+            partyStartIdx = 0;
+          }
+        }
+        const partyStartDir = dirs[partyStartIdx] ?? 'unknown';
+
+        if (!partyStartDir || !rotationPath)
```

Comment:
Can you check undefined here? (Or, can you check over the whole PR too and if there's places where you're looking for undefined, can you check that explicitly instead of a nullish check?)